app/src/main/java/bg/zahov/app/ui/history/HistoryScreen.kt

```diff
@@ -0,0 +1,236 @@
+                fontWeight = FontWeight.Bold
+            )
+            Text(
+                modifier = Modifier.padding(top = 4.dp),
+                text = item.date,
+                maxLines = 1,
+                overflow = TextOverflow.Ellipsis,
+                color = colorResource(R.color.less_vibrant_text),
+                style = MaterialTheme.typography.bodyLarge
+            )
+            Row(
+                Modifier
+                    .fillMaxWidth()
+                    .padding(top = 8.dp)
+                    .align(Alignment.CenterHorizontally),
+                horizontalArrangement = Arrangement.SpaceEvenly
+            ) {
+                TextWithLeadingIcon(
+                    text = item.duration,
+                    icon = painterResource(R.drawable.ic_clock),
+                    textModifier = Modifier
+                        .weight(1f)
+                        .padding(4.dp),
+                    iconColor = colorResource(R.color.less_vibrant_text),
+                    iconModifier = Modifier.align(Alignment.CenterVertically)
+                )
+                TextWithLeadingIcon(
+                    text = item.volume,
+                    icon = painterResource(R.drawable.ic_volume),
+                    textModifier = Modifier
+                        .weight(1f)
+                        .padding(4.dp),
+                    iconColor = colorResource(R.color.less_vibrant_text),
+                    iconModifier = Modifier.align(Alignment.CenterVertically)
+                )
+                TextWithLeadingIcon(
+                    text = item.personalRecords,
+                    icon = painterResource(R.drawable.ic_trophy),
+                    textModifier = Modifier
+                        .align(Alignment.CenterVertically)
+                        .weight(1f)
+                        .padding(4.dp),
+                    iconColor = colorResource(R.color.less_vibrant_text),
+                    iconModifier = Modifier.align(Alignment.CenterVertically)
+                )
+            }
+            Row(
+                modifier = Modifier
+                    .fillMaxWidth()
+                    .padding(top = 8.dp),
+                horizontalArrangement = Arrangement.SpaceBetween
+            ) {
+                Text(
+                    text = stringResource(R.string.exercise),
+                    color = Color.White,
+                    style = MaterialTheme.typography.bodyLarge
+                )
+                Text(
+                    text = stringResource(R.string.best_set),
+                    color = Color.White,
+                    style = MaterialTheme.typography.bodyLarge
+                )
+            }
+            val textFieldCount =
+                if (item.bestSets.size > 5 && item.exercises.size > 5) 5 else min(
+                    item.exercises.size,
+                    item.bestSets.size
+                )
+            for (i in 0 until textFieldCount) {
```

Comment:
I'd suggest using a LazyColumn with scroll disabled here.

Reply:
I do not think nesting lazyColumns is allowed without specifying height to the inner lazyColumn(app just crashes whenever we load with the second lazyColumn). I am not sure how to do it